Cesium.CodeGen/Ir/Expressions/IdentifierExpression.cs

```diff
@@ -33,10 +33,12 @@ public ILValue Resolve(IDeclarationScope scope)
     {
         scope.Variables.TryGetValue(Identifier, out var var);
         var par = scope.GetParameter(Identifier);
-
+        var global = scope.Context.GlobalType.Fields.FirstOrDefault(x => x.Name == Identifier);
```

Comment:
Does C allow global shadowing by a function parameter or a local variable?

If not, then you should also increase the `switch` here, and check if there's only one variable available (in the same manner it has been done for `(var, par)`.

Reply:
Yes, see 6.2.1.4 in  [C17 Language Standard Draft](http://www.open-std.org/jtc1/sc22/wg14/www/docs/n2310.pdf)